src/server/oicp/OICPUtils.ts

```diff
@@ -0,0 +1,211 @@
+import Constants from '../../utils/Constants';
+import OCPPStorage from '../../storage/mongodb/OCPPStorage';
+import { OICPAcknowledgment } from '../../types/oicp/OICPAcknowledgment';
+import { OICPEvseID } from '../../types/oicp/OICPEvse';
+import { OICPSession } from '../../types/oicp/OICPSession';
+import { ServerAction } from '../../types/Server';
+import Tenant from '../../types/Tenant';
+import Transaction from '../../types/Transaction';
+import TransactionStorage from '../../storage/mongodb/TransactionStorage';
+import moment from 'moment';
+
+export default class OICPUtils {
+
+  /**
+   * Return OICP Success Body Response
+   * @param {*} data
+   */
+  public static success(session: Partial<OICPSession>, data?: any): OICPAcknowledgment {
+    const response = {} as OICPAcknowledgment;
+    const statusCode = {} as OICPStatus;
+    statusCode.Code = OICPStatusCode.Code000;
+    response.StatusCode = statusCode;
+    response.Result = true;
+    response.EMPPartnerSessionID = session.empPartnerSessionID;
+    response.SessionID = session.id;
+    return response;
+  }
+
+  /**
+   * Return OICP no success Body Response
+   * @param {*} data
+   */
+  public static noSuccess(session: Partial<OICPSession>, data?: any): OICPAcknowledgment {
+    const response = {} as OICPAcknowledgment;
+    const status = {} as OICPStatus;
+    status.Code = OICPStatusCode.Code022;
+    status.Description = data;
+    response.StatusCode = status;
+    response.Result = false;
+    response.EMPPartnerSessionID = session.empPartnerSessionID;
+    response.SessionID = session.id;
+    return response;
+  }
+
+  /**
+   * Return OICP Error Body Response
+   * @param {*} error
+   */
+  public static toErrorResponse(error: Error): OICPAcknowledgment {
+    const response = {} as OICPAcknowledgment;
+    const status = {} as OICPStatus;
+    status.Code = OICPStatusCode.Code022;
+    status.Description = error.message;
+    response.StatusCode = status;
+    response.Result = false;
+    return response;
+  }
+
+  /**
+   * Build evse_id from charging station
+   * @param {*} countryCode the code of the CPO
+   * @param {*} partyId the partyId of the CPO
+   * @param {*} chargingStation the charging station used to build the evse ID
+   * @param {*} connector the connector used to build the evse id
+   */
+  public static buildEvseID(countryCode: string, partyId: string, chargingStation: ChargingStation, connector?: Connector): string {
+    let evseID = `${countryCode}*${partyId}*E${chargingStation.id}`;
+    if (!connector) {
+      for (const _connector of chargingStation.connectors) {
+        if (_connector) {
+          connector = _connector;
+          break;
+        }
+      }
+    }
+    evseID = `${evseID}*${connector.connectorId}`;
+    return evseID.replace(/[\W_]+/g, '*').toUpperCase();
+  }
+
+  public static breakUpEvseID(evseID: OICPEvseID): { countryCode: string, partyId: string, connectorId: string } {
```

Comment:
I don't get it.
you added chargingStationID, but I cannot get the chargingStationID from the evseID. 
I cannot derive the chargingStationID from evseID because all non alphanumeric characters and underscores are replaced with '\*' and all characters are in upper case.

I thought the number of the connector is the connectorID. Otherwise the type definition of the connector should be changed.

![image](https://user-images.githubusercontent.com/51398311/105467787-730cca80-5c96-11eb-9bd2-17867e97b957.png)

Reply:
> I don't get it.
> you added chargingStationID, but I cannot get the chargingStationID from the evseID.
> I cannot derive the chargingStationID from evseID because all non alphanumeric characters and underscores are replaced with '*' and all characters are in upper case.

You're right.
Then the EVSEID should be stored in the Charging Station oicpData.evse_id and used to find the charging station.

Reply:
> I thought the number of the connector is the connectorID. Otherwise the type definition of the connector should be changed.
I don't find the breakUpEvseID() any more so I cannot comment :(

Reply:
yes, I renamed it. Its called getEvseIdComponents() and I moved it to RoamingUtils